rust/src/metrics/reader.rs

```diff
@@ -0,0 +1,78 @@
+use crate::wit::wasi::otel::metrics::collect;
+use opentelemetry_sdk::{
+    error::{OTelSdkError, OTelSdkResult},
+    metrics::{
+        data::ResourceMetrics, reader::MetricReader, InstrumentKind, ManualReader, Temporality,
+    },
+};
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct WasiMetricReader {
```

Comment:
I know we're impling reader but I think it is more clear if we call this `WasiMetricExporter`. The spec suggests there is already a precedent (from pull style) of exporters being implemented as readers.